klemens_basic_thoughts_1.md

```diff
@@ -0,0 +1,206 @@
+# Design Proposal
+
+# Concepts
+
+For extensibility, we have two core concepts
+
+## ProcessLauncher
+
+The process Launcher is the class that starts the actual process: 
+
+```cpp
+template<typename T>
+concept bool ProcessLauncher() {
+    return requires(T launcher) {
+        {launcher.set_error(std::error_code())} -> void; //so initializers can post internal errors
+        {launcher()} -> process; //refine that so check tha parameter list, good enough for now.
+    };
+}
+```
+
+The reason to make this a concept is so replacement is possible. The use case in mind is `vfork` -> vfork is built into the posix-executor (renamed to launcher here) in boost-process and can be activated by a parameter. This is ugly and should not be in the standard; yet vfork as explicitly requested, and the possibility I see is to allow users to provide their own launcher OR the vendor to provide a `std::ext::vfork_launcher`. 
+
+Having this customization point does however allow more things, like security or virtualization.
```

Comment:
funny after reading this more carefully I like your reason for using concepts here -- we only constrain the important things and allow some non-portable flexibility for platform vendors.